lib/ansible/modules/cloud/azure/azure_rm_sql_database.py

```diff
@@ -0,0 +1,442 @@
+            - The name of the resource group that contains the resource. You can obtain this value from the Azure Resource Manager API or the portal.
+        required: True
+    server_name:
+        description:
+            - The name of the server.
+        required: True
+    name:
+        description:
+            - The name of the database to be operated on (updated or created).
+        required: True
+    tags:
+        description:
+            - Resource tags.
+        required: False
+    location:
+        description:
+            - Resource location.
+        required: True
+    collation:
+        description:
+            - The collation of the database. If createMode is not Default, this value is ignored.
+        required: False
+    create_mode:
+        description:
+            - "Specifies the mode of database creation.\n\nDefault: regular database creation.\n\nCopy: creates a database as a copy of an existing database.
+                sourceDatabaseId must be specified as the resource ID of the source database.\n\nOnlineSecondary/NonReadableSecondary: creates a database as
+               a (readable or nonreadable) secondary replica of an existing database. sourceDatabaseId must be specified as the resource ID of the existing p
+               rimary database.\n\nPointInTimeRestore: Creates a database by restoring a point in time backup of an existing database. sourceDatabaseId must
+               be specified as the resource ID of the existing database, and restorePointInTime must be specified.\n\nRecovery: Creates a database by restori
+               ng a geo-replicated backup. sourceDatabaseId must be specified as the recoverable database resource ID to restore.\n\nRestore: Creates a datab
+               ase by restoring a backup of a deleted database. sourceDatabaseId must be specified. If sourceDatabaseId is the database's original resource I
+               D, then sourceDatabaseDeletionDate must be specified. Otherwise sourceDatabaseId must be the restorable dropped database resource ID and sourc
+               eDatabaseDeletionDate is ignored. restorePointInTime may also be specified to restore from an earlier point in time.\n\nRestoreLongTermRetenti
+               onBackup: Creates a database by restoring from a long term retention vault. recoveryServicesRecoveryPointResourceId must be specified as the r
+               ecovery point resource ID.\n\nCopy, NonReadableSecondary, OnlineSecondary and RestoreLongTermRetentionBackup are not supported for DataWarehou
+               se edition. Possible values include: 'Copy', 'Default', 'NonReadableSecondary', 'OnlineSecondary', 'PointInTimeRestore', 'Recovery', 'Restore'
+               , 'RestoreLongTermRetentionBackup'"
+        required: False
+    source_database_id:
+        description:
+            - "Conditional. If createMode is Copy, NonReadableSecondary, OnlineSecondary, PointInTimeRestore, Recovery, or Restore, then this value is requir
```

Comment:
Split up as an entry per sentence instead of one long line, e.g.

```
source_database_id:
  description:
  - Specifies the resource ID of the source database.
  - If I(create_mode) is C(Copy), C(NonReadableSecondary), ...., then this must be set.
  - If I(create_mode) is C(NonReadableSecondary) or C(OnlineSecondary), the name of the source db must be the same as the new db.
```

Maybe even make it clearer whether this is the name of the database or the id, or both?